From 70ad21d39f097675a551575020b48684889a974f Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 26 Jan 2021 23:24:56 +0000
Subject: [PATCH] gnu: Add modularize-interfaces

* gnu/packages/lisp-xyz.scm
  (sbcl-modularize-interfaces cl-modularize-interfaces, ecl-modularize-interfaces):
  New variables
---
 gnu/packages/lisp-xyz.scm | 37 +++++++++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 12cd9df869..d80d0ca986 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -9823,6 +9823,43 @@ This package produces 1 system: @code{MODULARIZE-HOOKS}")
 (define-public cl-modularize-hooks
   (sbcl-package->cl-source-package sbcl-modularize-hooks))
 
+(define-public sbcl-modularize-interfaces
+  (let ((commit "96353657afb8c7aeba7ef5b51eb04c5ed3bcb6ef")
+        (revision "1"))
+    (package
+      (name "sbcl-modularize-interfaces")
+      (version (git-version "0.9.3" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/Shinmera/modularize-interfaces")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "0bjf4wy39cwf75m7vh0r7mmcchs09yz2lrbyap98hnq8blq70fhc"))))
+      (build-system asdf-build-system/sbcl)
+      (inputs
+       `(("modularize" ,sbcl-modularize)
+         ("trivial-indent" ,sbcl-trivial-indent)
+         ("trivial-arguments" ,sbcl-trivial-arguments)
+         ("lambda-fiddle" ,sbcl-lambda-fiddle)))
+      (home-page "https://shinmera.github.io/modularize-interfaces/")
+      (synopsis "Programmatical interfaces extension for Modularize")
+      (description
+       "This is an extension to @code{MODULARIZE} that allows your application to
+define interfaces in-code that serve both as a primary documentation and as
+compliance control.
+
+This package produces 1 system: @code{MODULARIZE-INTERFACES}")
+      (license license:zlib))))
+
+(define-public ecl-modularize-interfaces
+  (sbcl-package->ecl-package sbcl-modularize-interfaces))
+
+(define-public cl-modularize-interfaces
+  (sbcl-package->cl-source-package sbcl-modularize-interfaces))
+
 (define-public sbcl-moptilities
   (let ((commit "a436f16b357c96b82397ec018ea469574c10dd41"))
     (package
-- 
2.30.0

